File: csharp/ConversionHelper.cs
using System;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using System.Text;

namespace Ryu
{
    partial class Global //ConversionHelper
    {
        static int BooleanToInt32(bool value)
        {
            return value ? 1 : 0;
        }

        static uint BooleanToUInt32(bool value)
        {
            return value ? 1u : 0u;
        }

        static ulong BooleanToUInt64(bool value)
        {
            return value ? 1ul : 0ul;
        }

        static void memcpy(Span<sbyte> dest, sbyte[] src, uint srcOffset, int len)
        {
            src.AsSpan().Slice((int)srcOffset, len).CopyTo(dest);
        }

        static void memcpy(Span<sbyte> dest, sbyte[] src, int srcOffset, int len)
        {
            src.AsSpan().Slice(srcOffset, len).CopyTo(dest);
        }
    }
}

File: visualbasic/ConversionHelper.vb
Module ConversionHelper
    Function BooleanToInt32(value As Boolean) As Integer
        Return If(value, 1, 0)
    End Function

    Function BooleanToUInt32(value As Boolean) As UInteger
        Return If(value, 1UI, 0UI)
    End Function

    Function BooleanToUInt64(value As Boolean) As ULong
        Return If(value, 1UL, 0UL)
    End Function

    <Obsolete("Types with embedded references are not supported in this version of your compiler.")>
    Sub memcpy(dest As Span(Of Char), src() As Char, srcOffset As UInteger, len As Integer)
        src.AsSpan().Slice(CInt(srcOffset), len).CopyTo(dest)
    End Sub

    <Obsolete("Types with embedded references are not supported in this version of your compiler.")>
    Sub memcpy(dest As Span(Of Char), src() As Char, srcOffset As Integer, len As Integer)
        src.AsSpan().Slice(srcOffset, len).CopyTo(dest)
    End Sub
End Module
